Tools/PostProcessing/plot_nci_growth_rate.py

```diff
@@ -0,0 +1,185 @@
+                                dims=ds.domain_dimensions)
+    data = all_data[field].v.squeeze()
+    return data
+
+if __name__ == '__main__':
+
+    #--------------------------------------------------------------------------
+    # Users can set the input parameters defined in the main environment here,
+    # providing the correct paths to the diagnostics data to be post-processed
+    # as well as the start and end time steps between which the growth rate is
+    # computed.
+    #--------------------------------------------------------------------------
+
+    # Path to standard diagnostics,
+    # path to reduced diagnostics,
+    # path to save output figures
+    path_diags         = './path_to_standard_diagnostics/'
+    path_diags_reduced = './path_to_reduced_diagnostics/'
+    path_save          = './path_to_save_output_figures/'
+
+    # Name of standard diagnostics
+    # and reduced diagnostics (for total electromagnetic field energy)
+    diag_name = 'diag'
+    diag_name_reduced = 'field_energy.txt'
+
+    # Start and end iteration to compute growth rate
+    iter_min = 500
+    iter_max = 1000
+
+    #--------------------------------------------------------------------------
+
+    # Get list of output files
+    fn_l = glob.glob(os.path.join(path_diags, diag_name + '*[0-9]'))
+    fn_l.sort()
+
+    # Measure diagnostics frequency from two plotfiles and
+    # add 1 if last iteration does not match diagnostics frequency
+    dfreq = (int((re.search(os.path.join(path_diags, diag_name + '(.*)'), fn_l[1])).group(1))
+            - int((re.search(os.path.join(path_diags, diag_name + '(.*)'), fn_l[0])).group(1)))
+    itmin = iter_min // dfreq
+    itmax = iter_max // dfreq
+
+    # Get list of data at itmin and itmax
+    # (type: yt.frontends.boxlib.data_structures.WarpXDataset)
+    ds_itmin = yt.load(fn_l[itmin])
+    ds_itmax = yt.load(fn_l[itmax])
+
+    # Get data
+    field = ('boxlib', 'Ez')
+    data_itmin = load_data(ds_itmin, field)
+    data_itmax = load_data(ds_itmax, field)
+
+    # Get parameters from first data set
+    # Dt is the time between the min and max iteration, not the time step dt
+    Dt = (ds_itmax.current_time - ds_itmin.current_time).to_ndarray().item()
+    dxz = ds_itmin.domain_width / ds_itmin.domain_dimensions
+    dx = dxz[0]
+    dz = dxz[1]
+    kxmax = np.pi / dx
+    kzmax = np.pi / dz
+    Nx = data_itmin.shape[0]
+    Nz = data_itmin.shape[1]
+
+    # Compute spectrum
+    spectrum_itmin = np.fft.fftshift(np.fft.fft2(data_itmin))[Nx//2:,Nz//2:]
+    spectrum_itmax = np.fft.fftshift(np.fft.fft2(data_itmax))[Nx//2:,Nz//2:]
+
+    # Compute growth rate
+    growth_rate = np.log(np.abs(spectrum_itmax)) - np.log(np.abs(spectrum_itmin))
+    growth_rate /= c*Dt
+
+    # Plot energy over time (based on reduced diagnostics, if available)
+    energy = np.loadtxt(os.path.join(path_diags_reduced, diag_name_reduced), usecols=2)
+    fig, ax = plt.subplots(dpi=100)
+    ax.plot(energy, '-')
```

Comment:
Would it be possible here also to indicate time steps used for NCI growth rate calculation? Foe example, one can add something like this:
`ax.scatter([iter_min,iter_max], [energy[iter_min],energy[iter_max]], color='red')`